integration_tests/plugins/test_jhbuild_plugin.py

```diff
@@ -0,0 +1,97 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import http.client
+import http.server
+import glob
+import os
+import threading
+import integration_tests
+
+
+class StoppableHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
```

Comment:
Take a look at how we deal with fake servers in fixtures:
https://github.com/snapcore/snapcraft/blob/master/snapcraft/tests/fixture_setup.py#L253

You can reuse some things there, instead of implementing the quit in tear down yourself.